samples/f30-bootpart-hybrid.json

```diff
@@ -0,0 +1,150 @@
+{
+  "runner": "org.osbuild.fedora30",
+  "stages": [
+    {
+      "name": "org.osbuild.dnf",
+      "options": {
+        "releasever": "30",
+        "basearch": "x86_64",
+        "install_weak_deps": true,
+        "repos": [
+          "sha256:9f596e18f585bee30ac41c11fb11a83ed6b11d5b341c1cb56ca4015d7717cb97"
+        ],
+        "packages": [
+          "@Fedora Cloud Server",
+          "chrony",
+          "kernel",
+          "selinux-policy-targeted",
+          "grub2-pc",
+          "spice-vdagent",
+          "qemu-guest-agent",
+          "xen-libs",
+          "langpacks-en",
+          "grub2-efi-ia32",
+          "shim-ia32",
+          "grub2-efi-x64",
+          "shim-x64",
+          "efibootmgr",
+          "grub2-tools"
+        ],
+        "exclude_packages": [
+          "dracut-config-rescue"
+        ]
+      }
+    },
+    {
+      "name": "org.osbuild.systemd",
+      "options": {
+        "enabled_services": [
+          "cloud-config",
+          "cloud-final",
+          "cloud-init",
+          "cloud-init-local"]
+      }
+    },
+    {
+      "name": "org.osbuild.locale",
+      "options": {
+        "language": "en_US"
+      }
+    },
+    {
+      "name": "org.osbuild.fstab",
+      "options": {
+        "filesystems": [
+          {
+            "uuid": "08a41983-3328-4d1a-ae3c-1699e6d2a806",
+            "vfs_type": "ext4",
+            "path": "/boot",
+            "freq": "1",
+            "passno": "1"
+          },
+          {
+            "uuid": "7acfe2cc-4134-482a-a9d4-4979a5a87569",
+            "vfs_type": "ext4",
+            "path": "/",
+            "freq": "1",
+            "passno": "1"
+          },
+          {
+            "uuid": "46BB-8120",
+            "vfs_type": "vfat",
+            "path": "/boot/efi",
+            "options": "umask=0077,shortname=winnt",
+            "freq": "0",
+            "passno": "2"
+          }
+        ]
+      }
+    },
+    {
+      "name": "org.osbuild.grub2",
+      "options": {
+        "root_fs_uuid": "7acfe2cc-4134-482a-a9d4-4979a5a87569",
+        "boot_fs_uuid": "08a41983-3328-4d1a-ae3c-1699e6d2a806",
+        "kernel_opts": "ro biosdevname=0 net.ifnames=0 console=ttyS0 console=tty1",
+        "uefi": {
+          "vendor": "fedora"
+        },
+        "legacy": "i386-pc"
+      }
+    },
+    {
+      "name": "org.osbuild.fix-bls",
+      "options": {
+        "prefix": "/"
+      }
+    },
+    {
+      "name": "org.osbuild.selinux",
+      "options": {
+        "file_contexts": "etc/selinux/targeted/contexts/files/file_contexts"
+      }
+    }
+  ],
+  "assembler": {
```

Comment:
Maybe a separate discussion: Should/do we set up a protective MBR partition (and also possible the strange empty bootable partition hack you linked to) implicitly, or do we want to allow the MBR to be configured at the same time as the GPT, so the caller must set that up explicitly?

Reply:
`sfdisk` is already creating a protective MBR; inspect via `sfdisk -Y dos --json <disk>`:
```
{
   "partitiontable": {
      "label":"dos",
      "id":"0x00000000",
      "device":"/dev/sda",
      "unit":"sectors",
      "partitions": [
         {"node":"/dev/sda1", "start":1, "size":6291455, "type":"ee"}
      ]
   }
}
```
What we could do is add the [hack](https://wiki.archlinux.org/index.php/partitioning#Tricking_old_BIOS_into_booting_from_GPT) to trick the old BIOS into booting from GPT.